c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -3676,6 +3681,133 @@ bool response_started(JSObject *self) {
 }
 } // namespace FetchEvent
 
+using std::chrono::system_clock;
+using TimerArgumentsVector = std::vector<JS::Heap<JS::Value>>;
+
+class Timer {
+public:
+  uint32_t id;
+  JS::Heap<JSObject *> callback;
+  TimerArgumentsVector arguments;
+  uint32_t delay;
+  system_clock::time_point deadline;
+  bool repeat;
+
+  Timer(uint32_t id, HandleObject callback, uint32_t delay, system_clock::time_point deadline,
+        JS::HandleValueVector args, bool repeat)
+      : id(id), callback(callback), delay(delay), deadline(deadline), repeat(repeat) {
+    deadline = system_clock::now() + system_clock::duration(delay * 1000);
+    arguments.reserve(args.length());
+    for (auto arg : args) {
+      arguments.push_back(JS::Heap(arg));
+    }
+  }
+
+  friend bool operator<(Timer const &left, Timer const &right) {
+    return left.deadline < right.deadline;
+  }
+
+  void trace(JSTracer *trc) {
+    JS::TraceEdge(trc, &callback, "Timer callback");
+    for (auto &arg : arguments) {
+      JS::TraceEdge(trc, &arg, "Timer callback arguments");
+    }
+  }
+};
+
+class ScheduledTimers {
+public:
+  Timer *first() { return *timers.begin(); }
+
+private:
+  std::list<Timer *> timers;
+  static uint32_t next_id;
+
+  uint32_t add_timer(Timer *timer) {
+    auto iter = timers.begin();
+
+    for (; iter != timers.end(); iter++) {
+      if ((*iter)->deadline > timer->deadline) {
+        break;
+      }
+    }
+
+    auto inserted = (*timers.insert(iter, timer));
+    return inserted->id;
+  }
+
+  void repeat_first() {
+    Timer *timer = first();
```

Comment:
What happens if the `timers` list is empty? If this function won't be called in that case, it would be worth adding a comment here explaining why a null check isn't necessary.

Reply:
A comment would be good here, thanks!